open_humans/account_views.py

```diff
@@ -114,11 +106,18 @@ class MemberChangeEmailView(PrivateMixin, AccountSettingsView):
         },
     }
 
-    def get_success_url(self, *args, **kwargs):
-        kwargs.update(
-            {'fallback_url': reverse_lazy('my-member-settings')})
-        return super(MemberChangeEmailView, self).get_success_url(
-            *args, **kwargs)
+    def post(self, request, *args, **kwargs):
+        form = self.get_form()
+        email = str(form.data['email'])
+        emailaddress = EmailAddress.objects.filter(email=email)
+        if emailaddress.count() == 1:
+            if emailaddress.first().primary == False:
+                emailaddress.delete()
+
+        if form.is_valid():
+            self.form_valid(form)
+            return super().post(request, *args, **kwargs)
```

Comment:
Why not `return self.form_valid(form)`?

AIUI the output of that is expected to be a response, and I think the parent is ultimately calling on the following via inheritance from `FormView`: https://github.com/django/django/blob/master/django/views/generic/edit.py#L57

Reply:
good point.  I still think that I should call super, though, just to get things like get_success_url() called.

Reply:
oh, you're right.  Killing the super()